pkg/limayaml/validate.go

```diff
@@ -647,3 +661,39 @@ func ValidateAgainstLatestConfig(yNew, yLatest []byte) error {
 
 	return nil
 }
+
+func validateSocket(engine, socket string) error {
+	if socket == "" {
+		return fmt.Errorf("%s socket path must not be empty", engine)
```

Comment:
The normal format of validation errors would look more like

```
field `portMonitor.docker.sockets[1]` must not be empty
```

You will need to pass in the index too.

Reply:
The `validateSocket` gets called in a loop, so no need for the index. The socket is individual and not a list.

Reply:
No, the sockets are a list, so you should provide the list index for the error message. You can pass in the field name instead of the index if you want the caller to assemble it. But since it is called from 2 places, it would be simpler to pass in the index.

```suggestion
		return fmt.Errorf("field `portMonitor.%s.sockets[%d]` must not be empty", engine, i)
```